src/kafunk/Consumer.fs

```diff
@@ -390,6 +390,20 @@ type ConsumerMessageSet =
     static member messages (ms:ConsumerMessageSet) =
       ms.messageSet.messages |> Array.map (fun x -> x.message)
 
+///Range that contains start offsets and end offsets used by consume range
+type OffsetRange = 
+  { startOS : Offset []
```

Comment:
do we need to make the constructor private given that there are accessor methods for Start and End?

Reply:
I'd say we get rid of this type entirely, or at least make it private. We ought to keep the public namespace as succinct as possible.